met-api/src/met_api/services/engagement_metadata_service.py

```diff
@@ -145,6 +138,37 @@ def update(metadata_id: int, value: str) -> dict:
         metadata.value = value
         return dict(EngagementMetadataSchema().dump(metadata, many=False))
 
+    @staticmethod
+    @transactional()
+    def update_by_taxon(engagement_id: int, taxon_id: int, values: List[str]) -> List[dict]:
+        """
+        Update engagement metadata by taxon.
+
+        Args:
+            engagement_id: The ID of the engagement.
+            taxon_id: The ID of the metadata taxon.
+            values: The values to store for the taxon.
+        Returns:
+            The updated metadata as a list.
+        """
+        query = EngagementMetadata.query.filter_by(
+            engagement_id=engagement_id, taxon_id=taxon_id)
+        metadata = query.all()
+        if len(metadata) > len(values):
```

Comment:
This is my attempt to minimize the amount of new EngagementMetadata models stored to the database. By overwriting their values, it prevents incrementing the auto-increment ID field. However, for data semantics this might not actually be the best (metadata IDs don't always refer to the same value). I can change the implementation quickly if necessary